src/blocks/block-coauthors/edit.js

```diff
@@ -0,0 +1,263 @@
+		<div
+			{ ...useInnerBlocksProps(
+				{ className: 'wp-block-co-authors-plus-coauthor' },
+				{ template: [ [ 'co-authors-plus/name' ] ] }
+			) }
+		/>
+	);
+}
+
+const ALLOWED_FORMATS = [ 'core/bold', 'core/italic', 'core/text-color' ];
+
+/**
+ * The edit function describes the structure of your block in the context of the
+ * editor. This represents what the editor will render when the block is used.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#edit
+ *
+ * @return {WPElement} Element to render.
+ */
+export default function Edit( {
+	attributes,
+	setAttributes,
+	clientId,
+	context,
+	isSelected,
+} ) {
+	const { prefix, separator, lastSeparator, suffix, layout, textAlign } =
+		attributes;
+	const { postId } = context;
+	const authorPlaceholder = useSelect(
+		( select ) => select( 'co-authors-plus/blocks' ).getAuthorPlaceholder(),
+		[]
+	);
+	const [ coAuthors, setCoAuthors ] = useState( [ authorPlaceholder ] );
+	const [ activeBlockContextId, setActiveBlockContextId ] = useState();
+	const noticesDispatch = useDispatch( 'core/notices' );
+
+	useEffect( () => {
+		if ( ! postId ) {
+			return;
+		}
+
+		const controller = new AbortController();
+
+		apiFetch( {
+			path: `/coauthors/v1/coauthors?post_id=${ postId }`,
+			signal: controller.signal,
+		} )
+			.then( setCoAuthors )
+			.catch( handleError );
+
+		return () => {
+			controller.abort();
+		};
+	}, [ postId ] );
+
+	/**
+	 * Handle Error
+	 *
+	 * @param {Error}
+	 */
+	function handleError( error ) {
+		if ( 'AbortError' === error.name ) {
+			return;
+		}
+		noticesDispatch.createErrorNotice( error.message, {
+			isDismissible: true,
+		} );
+	}
+
+	const blocks = useSelect( ( select ) => {
+		return select( blockEditorStore ).getBlocks( clientId );
+	} );
+
+	const setLayout = ( nextLayout ) => {
+		setAttributes( {
+			layout: { ...layout, ...nextLayout },
+		} );
+	};
+
+	const layoutControls = [
+		{
+			icon: list,
+			title: __( 'Inline', 'co-authors-plus' ),
+			onClick: () => setLayout( { type: 'inline' } ),
+			isActive: layout.type === 'inline',
+		},
+		{
+			icon: grid,
+			title: __( 'Block', 'co-authors-plus' ),
+			onClick: () => setLayout( { type: 'block' } ),
+			isActive: layout.type === 'block',
+		},
+	];
+
+	return (
+		<>
+			<BlockControls>
+				<ToolbarGroup controls={ layoutControls } />
+				<AlignmentControl
+					value={ textAlign }
+					onChange={ ( nextAlign ) => {
+						setAttributes( { textAlign: nextAlign } );
+					} }
+				/>
+			</BlockControls>
+			<div
+				{ ...useBlockProps( {
+					className: classnames( {
+						[ `is-layout-cap-${ layout.type }` ]: layout.type,
+						[ `has-text-align-${ textAlign }` ]: textAlign,
+					} ),
+					style: {
+						gap: __experimentalGetGapCSSValue(
+							attributes?.style?.spacing?.blockGap
+						),
+					},
+				} ) }
+			>
+				{ coAuthors &&
+					'inline' === layout.type &&
+					( isSelected || prefix ) && (
+						<RichText
```

Comment:
I am seeing this too. What I'm experiencing is that if I take out the default prefix `By ` in the site editor I cannot click back into the richtext field to change it. I'm not having this same problem while editing a post or page.

Reply:
@laurelfulford I thought I was experiencing the same issue, but maybe I was wrong. I thought I was unable to edit the prefix in the site editor, but it turned out it was just hard to see the cursor because of a box-shadow in the focus and hover state styles.

- Is this the same issue you were seeing?
  - If so, does knowing that this works the same as the core paragraph block resolve the issue?
  - If not, can you provide some additional context so I can look into it?

When editing a post, I am able to click on the prefix and I can see the cursor.
<img width="629" alt="Screenshot 2024-01-16 at 5 56 51 PM" src="https://github.com/Automattic/Co-Authors-Plus/assets/226381/5e1b40cb-8fcf-4615-a191-d3d08d2de29c">

When editing the theme, it seems like I can't click on the prefix because the cursor never appears. However I am able to type.
<img width="422" alt="Screenshot 2024-01-16 at 5 57 31 PM" src="https://github.com/Automattic/Co-Authors-Plus/assets/226381/c3d29699-b613-4c11-93f7-4f45ad782387">

The cursor is actually there, it is hidden behind a box-shadow. If I manually remove the box-shadow, the cursor is present.
<img width="512" alt="Screenshot 2024-01-16 at 5 58 53 PM" src="https://github.com/Automattic/Co-Authors-Plus/assets/226381/1694220e-590d-4674-88a6-56201a6d336c">

There are additional states where the box-shadow is applied while using in the site editor. Here's the same block with the CSS shown in the inspector. It seems to be related to `outlineMode` [as shown here in the editor settings](https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/provider/use-block-editor-settings.js#L272).
<img width="1162" alt="Screenshot 2024-01-16 at 5 59 43 PM" src="https://github.com/Automattic/Co-Authors-Plus/assets/226381/d1c5e129-6adc-4302-9164-54f5fe2a2178">

Here is the paragraph block showing the same behavior.
<img width="288" alt="Screenshot 2024-01-16 at 6 00 30 PM" src="https://github.com/Automattic/Co-Authors-Plus/assets/226381/56d475d3-ab17-41a1-9955-4d8b30dc24e5">

Reply:
I looked closer at the screenshot of the inspector in my previous comment, and saw the use of the `remove-outline` class. I added that to the Co-Authors Block in the site editor and I am now able to see the cursor correctly.

Does https://github.com/thoughtis/Co-Authors-Plus/commit/028e14aa7462ebd3da3a15e18a6ed58d28bace98 resolve this issue?

<img width="409" alt="Screenshot 2024-01-16 at 6 43 21 PM" src="https://github.com/Automattic/Co-Authors-Plus/assets/226381/14d775e3-8860-49b7-b6d6-c50f02ff4ee8">

Reply:
Yes, it makes a big difference -- thanks!